src/storage/storage_structure/lists/lists_update_store.cpp

```diff
@@ -88,6 +102,46 @@ void ListsUpdateStore::insertRelIfNecessary(shared_ptr<ValueVector>& srcNodeIDVe
     }
 }
 
+void ListsUpdateStore::deleteRelIfNecessary(const shared_ptr<ValueVector>& srcNodeIDVector,
+    const shared_ptr<ValueVector>& dstNodeIDVector, const shared_ptr<ValueVector>& relIDVector) {
+    auto srcNodeID =
+        srcNodeIDVector->getValue<nodeID_t>(srcNodeIDVector->state->getPositionOfCurrIdx());
+    auto dstNodeID =
+        dstNodeIDVector->getValue<nodeID_t>(dstNodeIDVector->state->getPositionOfCurrIdx());
+    auto relID = relIDVector->getValue<int64_t>(relIDVector->state->getPositionOfCurrIdx());
+    auto tupleIdx = isRelStoredInUpdateStore(relID);
```

Comment:
Let's be careful about the terminology of "insert, update, delete". We use "update" to mean any type of modification and this is fine. But when you mean insert somewhere, be explicit about it. For example the below comment is confusing because you talk about being in *update*Store but then about *inserted*RelsTupleIdxInFT. If you want to be explicit about the rel being inserted, say insert. 

Also, I notice that you don't remove the rel from the factorizedTable. So we leak some space and don't reuse it. You might want to add a note here being explicit that we leak space and don't reuse it. Down the road this may be a problem.

Reply:
1. Updated the comment to: 
```
// If the rel that we are going to delete is a newly inserted rel, we simply need to delete
        // its tupleIdx from the insertedRelsTupleIdxInFT of listUpdateStore in FWD and BWD
        // direction.
```
And update： `isRelStoredInUpdateStore`-> `isNewlyInsertedRel`
2.  Added a comment saying that we leak space and don't reuse it in the factorizedTable.